koji-offsite-backup/osg-koji-offsite-backup

```diff
@@ -224,26 +225,48 @@ def get_previous_remote_backup_date(current_backup_date):
 
     # return dirs in reverse sorted order
     # dates are in YYYY-mm-dd format so they sort
-    ret, backup_dirs = remote_run_command('cd %s; ls -dr */' % REMOTE_BASE_PATH)
+    ret, backup_dirs = remote_run_command('cd "%s" && ls -dr 20[0-9][0-9]-[0-1][0-9]-[0-3][0-9]/' % REMOTE_BASE_PATH)
     if ret != 0:
         return None
 
-    date_re = re.compile('20[0-9][0-9]-[0-1][0-9]-[0-3][0-9]')
-
-    backup_dates = [x.rstrip('/') for x in backup_dirs.split('\n') if date_re.match(x)]
+    backup_dirs = backup_dirs.strip()
+    backup_dates = [x.rstrip('/') for x in backup_dirs.split('\n') if x]
     for date in backup_dates:
         if date < current_backup_date:
             return date
     else:
         return None
 
 
+def get_current_backup_date():
+    """Return the most recent completed backup date"""
+
+    ret, backup_dirs = sbacktick('cd "%s" && ls -dr 20[0-9][0-9]-[0-1][0-9]-[0-3][0-9]/' % LOCAL_BACKUP_ROOT, shell=True)
+
+    if ret != 0:
+        return None
+
+    backup_dirs = backup_dirs.strip()
+    backup_dates = [x.rstrip('/') for x in backup_dirs.split('\n') if x]
+    ret, _ = sbacktick("ps axw | grep -q [r]sync", shell=True)
```

Comment:
To me this `ps axw | grep -q [r]sync` looks like trouble...

Besides the fact that it actually breaks if there is a file named `rsync` in the current directory (that is, `grep` would always succeed, as it will match its own entry from `ps` in that case), even if `[r]sync` were quoted properly it would still match any other processes with the string `rsync` anywhere in the command line.

You could test the condition of `rsync` running it a bit more directly with something like `pgrep -x rsync >/dev/null`, but there is still the question of whether you care about other rogue `rsync` processes unrelated to `osg-koji-offsite-backup` ... you could do `pgrep -x rsync -u root >/dev/null` to at least exclude user `rsync` processes from interfering ... but the problem could still be there unless you have a way to identify an `rsync` process as belonging to another instance of `osg-koji-offsite-backup` :-\

Does it matter?  (I don't really know; what do you think?)

Or would it work/make sense to search instead for another `osg-koji-offsite-backup` running?